src/disk_io_thread.cpp

```diff
@@ -193,63 +197,25 @@ namespace libtorrent
 	void disk_io_thread::abort(bool wait)
 	{
 		m_abort = true;
-		if (m_num_threads == 0)
+		if (num_threads() == 0)
 		{
 			abort_jobs();
 		}
 		else
 		{
-			set_num_threads(0, wait);
+			m_generic_threads.abort(wait);
```

Comment:
As far as can tell this code duplicates the existing behavior. Both before and after these changes the last worker thread will keep running until the job queue is empty.